### PR TITLE
[SPARK-35310][MLLIB] Update to breeze 1.2

### PR DESCRIPTION
**Pull request by srowen**
### What changes were proposed in this pull request?

Update to the latest breeze 1.2

### Why are the changes needed?

Minor bug fixes

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

Existing tests

### CODE REVIEW

File: mllib/src/test/scala/org/apache/spark/ml/optim/WeightedLeastSquaresSuite.scala [hunk 1]
```diff
@@ -531,7 +534,8 @@ class WeightedLeastSquaresSuite extends SparkFunSuite with MLlibTestSparkContext
          standardization <- Seq(false, true)) {
       for (solver <- WeightedLeastSquares.supportedSolvers) {
         val wls = new WeightedLeastSquares(fitIntercept, regParam, elasticNetParam = 0.0,
-          standardizeFeatures = standardization, standardizeLabel = true, solverType = solver)
+          standardizeFeatures = standardization, standardizeLabel = true, solverType = solver,
```
Comment (srowen) [inline mllib/src/test/scala/org/apache/spark/ml/optim/WeightedLeastSquaresSuite.scala:537]: The updated tol is needed to allow it to pass - and matches the tol in the R code that was used to generate 'correct' answers, so seems OK. Same with maxIter. I made a similar change throughout this test suite for consistency.

File: mllib/src/test/scala/org/apache/spark/mllib/linalg/VectorsSuite.scala [hunk 2]
```diff
@@ -295,7 +295,9 @@ class VectorsSuite extends SparkFunSuite with Logging {
       val denseVector1 = Vectors.dense(sparseVector1.toArray)
       val denseVector2 = Vectors.dense(sparseVector2.toArray)
 
-      val squaredDist = breezeSquaredDistance(sparseVector1.asBreeze, sparseVector2.asBreeze)
+      val squaredDist = sparseVector1.toArray.zip(sparseVector2.toArray).map {
```
Comment (srowen) [inline mllib/src/test/scala/org/apache/spark/mllib/linalg/VectorsSuite.scala:298]: There was a weird compile error on breeze's squaredDistance, so I just wrote it out here instead

**Comment by HyukjinKwon** [comment]
Just a reminder, we should probably either run the dependency test manually for dependency changes, or wait for GitHub Actions to be fixed .. Jenkins skips dependency tests for now due to the unexpected issues in the machines (https://github.com/apache/spark/commit/0e33b5ecde12452cbd5e5f4c54bf0b6388b502bd)

**Comment by srowen** [comment]
OK sounds good. These changes were generated by the script, so should be OK.

**Review by HyukjinKwon — APPROVED** [review]

**Review by dongjoon-hyun — APPROVED** [review]

**Comment by srowen** [comment]
The last test passed; I only force-pushed to fix a deps file conflict, which also has passed. I'll merge to master

**Comment by pan3793** [comment]
Some of the transitive deps of breeze 1.0 look like not a stable version, i.e. `algebra_2.12-2.0.0-M2.jar`, `cats-kernel_2.12-2.0.0-M4.jar`, `spire_2.12-0.17.0-M1.jar`, breeze 1.2 fixed this issue. I don't think it's a good idea to shape the milestone version deps in a release, does this PR have a chance to enter release 3.2?

**Comment by srowen** [comment]
Heh, it's been like that for many releases, and I don't know of any issues.
While I might not back-port to a maintenance branch, 3.2.0 seems reasonable. There's the possibility of a small change in some MLlib outputs (see test change) because of a fix in breeze 1.2.
If there aren't strong objections, yeah, maybe that's fine.

**Comment by srowen** [comment]
Merged to 3.2 as well

**Pull request closed without merging**